### ISSUE TITLE
Include module names in the log

### ISSUE THREAD

**Issue by meetmangukiya** [coala/coala#3557]
Currently logging is done **directly** like:
```py
import logging
logging.debug()
logging.warning()
```
leading to logs looking like :
```
DEBUG:root: ...
WARNING:root: ...
```
Instead, logs looking like this:
```
WARNING: coalib.bearlib.abstractions.linter: Use of capture groups is discouraged
```
should be adapted :thinking: 

area/logging status/needs design

**Comment by Makman2** [comment]
like it :+1:
is there a flag in `logging` to enable this behaviour?

**Comment by meetmangukiya** [comment]
It is fairly easy to do(I may need to dive deep), but little search gave me:
```
logger = logging.getLogger(__name__)
logger.debug('log')
```
outputs:
```
DEBUG:package.subpackage.module: log
```
which makes me wonder why aren't we using it this way already :sweat_smile:

**Comment by Makman2** [comment]
ahhh we need to get a logger instance manually in every file... that's bad... is there some automatic way?

**Comment by meetmangukiya** [comment]
Why is that bad?

**Comment by Makman2** [comment]
then we need to write those two lines everywhere + our API encourages already to use `logging` directly.

**Comment by meetmangukiya** [comment]
So, have we reached the conclusion of wontfix ?

**Comment by Makman2** [comment]
no we need a better approach :)

**Comment by nemaniarjun** [comment]
This is a non issue. We dont print logs as mentioned in the original post. 
Our logs are formatted as this: 
```
[WARNING][21:03:52] The default coafile '.coafile' was not found. You can generate a configuration file with your current options by adding the `--save` flag or suppress any use of config files with `-I`.
[WARNING][21:03:52] coala will not run any analysis. Did you forget to give the `--files` argument?
[WARNING][21:03:52] coala will not run any analysis. Did you forget to give the `--bears` argument?
```

**Comment by nemaniarjun** [comment]
We can easily add the logger module name, but do we need it?

**Comment by Makman2** [comment]
Oh indeed. Though it doesn't harm to pass the module information to `logging` if possible. But if we don't find something better than `logger = logging.getLogger(__name__)`, I'm also voting for a wontfix^^

**Comment by meetmangukiya** [comment]
I'd like to bring into notice that these logs :point_up: are printed by the *hybrid* log_printer of coala...  We do have loggings like `logging.warning`, you can git grep it. And these warnings are actually for the user and not the developer, while I really think that adding the module names to the debug logs will be really useful, I still vote for not a wontfix and further discussion...

**Comment by Makman2** [comment]
We need something better, it's not feasible currently to do everywhere `logger = ...` and use `logger` instead of `logging` directly.

**Comment by meetmangukiya** [comment]
Sure, I was just trying to defend my point and this issue from won't fix :)

**Comment by Makman2** [comment]
> I'm also voting for a wontfix^^

I see... I wanted to say

> But if we don't find something better than logger = logging.getLogger(__name__), **then** I'm also voting for a wontfix^^

:3

**Comment by Mixih** [comment]
have we looked into logformat objects?

**Comment by nemaniarjun** [comment]
Its very easy to implement this once we use the updated Logging. 
We just need to add `pathname` or `module` to the colourlog formatter. ( Tested that right now ) 
But that would just print out `LogPrinter`, since thats where the messages are originated rn.

**Comment by nemaniarjun** [comment]
We can also have different formats for different levels of log. Something that `colorlog` (the third party formatter we use) supports.

**Comment by Makman2** [comment]
I think the problem is to get the actual file/module. As soon as we call `logging.log`, `logging.warning` the module is different. If we set the module in the beginning, then it stays this module. I can only imagine to (mis)use traceback information somehow, and then to modify the function responsible for formatting to use and process this information^^

**Comment by nemaniarjun** [comment]
@Makman2 So this can be blocked depending on the timeline of removing `log_printer` completely.
Its not worth doing this, if we are going to use direct logging anyway.

**Comment by Makman2** [comment]
@nemaniarjun imo this issue is not dependent on the `log_printer` removal, currently `log_printer` is deprecated and uses `logging` internally^^

**Comment by nemaniarjun** [comment]
If this is critical we can implement it using traceback checking rn (diff/medium)
But if we wait till we have completely removed `log_printer` we just need to change the format we are printing the log in. (less than diff/low)

**Comment by Makman2** [comment]
It's not critical. But the problem is not to change the format (you could already do that in `configure_logging`), it's that it would display just `root` and not the correct file/module.

**Comment by nemaniarjun** [comment]
No! If we change the format, it can display the module name where the log originated from.
Right now all log messages are originated in `log_printer` so thats what is printed if you add `module` to the format.

**Comment by Makman2** [comment]
Ah really? Then we should wait for the complete removal of `log_printer`.

Blocked by https://github.com/coala/coala/issues/975

**Comment by meetmangukiya** [comment]
@nemaniarjun it doesn't print the module name **from** which log originated from, it prints the name of the logger object that created the log message, that one passes during construction. 

One has to create a root logger to get all the log messages, or set the parents of the logger appropriately.

**Comment by nemaniarjun** [comment]
@meetmangukiya I think you are confusing `module` and `name` attributes.
Documentation is here : https://docs.python.org/3.6/library/logging.html#logrecord-attributes
